### PR TITLE
feat: support multiple image push

### PR DESCRIPTION
**Pull request by iiian**
Hey @RafikFarhad! So, I'm digging this project, but it doesn't quite support my use-case, or at least, I think I'd need to get creative to make it work.

I'm in a monorepo context, and currently I've got one job building N images associated to N separate component services. Whether or not that's the "right" approach, it's currently my "fast" approach.

What the change I am proposing would do is introduce a boolean flag `all_prefixed_images`, which, when set, scrapes the local `docker` daemon for everything that has been created and is bound for your $registry/$project-id domain. *Those* images are then slated for push.

This is an early-stage draft, my testing completion status is pretty minimal right now:
[X] verify locally with the `docker images --filter --format` stuff & env variables
[ ] build a real test case

I'm mostly panning to see whether or not you want this in your repo.

Let me know, and keep up the awesome work!

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -83,24 +83,36 @@ else
...
+else
+    IMAGE_NAMES=($INPUT)
+fi
 
-    IMAGE_NAME="$INPUT_REGISTRY/$INPUT_PROJECT_ID/$INPUT_IMAGE_NAME:$IMAGE_TAG"
+for IMAGE_NAME in ${IMAGE_NAMES[@]}; do
+    for IMAGE_TAG in ${ALL_IMAGE_TAG[@]}; do
 
-    echo "Fully qualified image name: $IMAGE_NAME"
+        IMAGE_NAME="$INPUT_REGISTRY/$INPUT_PROJECT_ID/$IMAGE_NAME:$IMAGE_TAG"
```
Comment (iiian) [inline entrypoint.sh:99]: shadow `$IMAGE_NAME` from line `96`

**Comment by RafikFarhad** [comment]
Hi, thanks for your contribution and detailed code reading. 
I am curious about one part, why can't we use `push_only: true` and provide the push-able image names to this action? Don't we know the image names at the point when we are calling this action?

**Comment by iiian** [comment]
Hey @RafikFarhad, you make a valid observation, and I think we could do as much. However, I think we _should_ instead close this PR, because I convinced myself out of the approach I'm suggesting here. I realized it was feasible and favorable to reorganize my automated container publication into concurrent calls to this GitHub action by relying on a dynamically populated job matrix.

Happy new year!

**Comment by RafikFarhad** [comment]
Hi @iiian, happy new year 🎆 !!! 

Let us know if this action is sufficient for your new approach. And feel free to close this PR when you are sure that this PR is no more needed for this action. Thanks for your contribution 🥂 .

**Comment by almereyda** [comment]
Thanks everyone your contemplation here. Before this conversation wanes, please allow to ask a last question, to @iiian:

What is *a dynamically populated job matrix* in your use of the term? I have a vague impression, but could also go well with little examples. How is such a job matrix laid out, and how does one populate so with *concurrent calls to this GitHub action*?

Any pointers will be highly appreciated.

**Comment by iiian** [comment]
@almereyda I've got a two-phase Github actions job that sits on the `master` branch, and listens for changes. 

The first phase diffs the new changes against the previous commit hash. Using scripts within the codebase, this phase yields a bash-list of monorepo subcomponent identifiers associated with images that need to be built and deployed to the Github actions runtime, via `$GITHUB_OUTPUT`. For instance, if you had 3 components in your monorepo, `front-end, service-a, service-b` then you might see `AFFECTED_PROJECTS=(front-end service-a)`.

The second phase uses `fromJSON` & `matrix` to bake N jobs, one for each affected project. See below for an abridged example `actions.yml`.

This way, we can skip building redundant projects per merge-to-master.

```yaml
name: Build and Deploy
on:
  push:
    branches:
      - master
jobs:
  detect-affected-components:
    runs-on: ubuntu-latest
    outputs:
      AFFECTED_PROJECTS: ${{ steps.compute.outputs.AFFECTED_PROJECTS }}
    steps:
      - name: Computing affected projects...
        id: compute
        run: |
          ... # run a handful of actions to compute the set of affect monorepo subcomponents
              # changes based on diff against previous state of master vs new state of master
          echo "AFFECTED_PROJECTS=$AFFECTED_PROJECTS" >> $GITHUB_OUTPUT
  build-and-push-to-gcr:
    needs: detect-affected-components
    strategy:
      matrix:
        component: ${{ fromJSON(needs.detect-affected-components.outputs.AFFECTED_PROJECTS) }}
    runs-on: ubuntu-latest
    steps:
      - run: |
          # build the image using scripts in codebase + ${{ matrix.component }}
      - name: Pushing to gcloud...
        uses: RafikFarhad/push-to-gcr-github-action@v4.1.0
        with:
          gcloud_service_key: ${{ secrets.GCLOUD_SERVICE_KEY }}
          registry: ...
          project_id: ...
          image_name: ${{ matrix.component }}
          push_only: true
```

**Pull request closed without merging**